libkbfs/bcache.go

```diff
@@ -238,6 +242,8 @@ func (b *BlockCacheStandard) PutWithPrefetch(
 		defer func() {
 			if madeRoom {
 				b.cleanTransient.Add(ptr.ID, blockContainer{block, hasPrefetched})
+			} else {
+				err = cachePutCacheFullError{ptr}
```

Comment:
Oh, I just realized that calling `Put` and `PutWithPrefetch` for the same block will double-count the size of the block in `cleanTotalBytes` (in `makeRoomForSize`)!  So if the block is already in the cache, we should probably subtract its size (and delete it from the cache?) before making room for it again.